### PR TITLE
Add Redpanda module

### PR DESCRIPTION
**Pull request by eddumelendez**
New module uses `docker.redpanda.com/vectorized/redpanda:v22.2.1` docker image as a
base giving the `--mode dev-container` flag added in that version.

Co-authored-by: Sergei Egorov <bsideup@gmail.com>

### CODE REVIEW

**Comment by eddumelendez** [comment]
inspired by https://gist.github.com/bsideup/ffe585b3d5c640f94d6771b5e850c8a9

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    protected void containerIsStarting(InspectContainerResponse containerInfo) {
+        super.containerIsStarting(containerInfo);
+
+        String command = "#!/bin/bash\n";
+
+        command += "/usr/bin/rpk redpanda start --mode dev-container ";
+        command += "--kafka-addr PLAINTEXT://0.0.0.0:29092,OUTSIDE://0.0.0.0:9092 ";
+        command += "--advertise-kafka-addr PLAINTEXT://kafka:29092,OUTSIDE://" + getHost() + ":" + getMappedPort(9092);
+
+        copyFileToContainer(Transferable.of(command.getBytes(StandardCharsets.UTF_8), 0777), STARTER_SCRIPT);
```
Comment (bsideup) [inline modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:51]:
```suggestion
        copyFileToContainer(Transferable.of(command, 0777), STARTER_SCRIPT);
```

File: docs/modules/redpanda.md [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Redpanda
+
+Testcontainers can be used to automatically instantiate and manage [Redpanda](https://redpanda.com/) containers.
+More precisely Testcontainers uses the official Docker images for [Redpanda](https://hub.docker.com/r/vectorized/redpanda/)
```
Comment (emaxerrno) [inline docs/modules/redpanda.md:4]: this works ok, we added a small translation via scarf.sh that allows us to point to a different backend while giving users aa stable name - this works today but would prevent us from moving to amazon or smth like that. 

just a nit, but would be easier to support via 
```
docker.redpanda.com/vectorized/redpanda:$VERSION 
```

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java (hunk 1 above)
Comment (eddumelendez) [reply modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:51]: there is no `Transferable.of(String, int)`

Comment (bsideup) [reply modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:51]: Ahhh.... Let's add one then 😀

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+    }
+
+    public RedpandaContainer(DockerImageName imageName) {
+        super(imageName);
+        imageName.assertCompatibleWith(REDPANDA_IMAGE);
+
+        boolean isLessThan22_2_1 = new ComparableVersion(imageName.getVersionPart()).isLessThan("v22.2.1");
+        if (isLessThan22_2_1) {
+            throw new IllegalArgumentException("Redpanda version must be >= v22.2.1");
+        }
```
[the remark this answers is not in the document]
Comment (eddumelendez) [reply modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java]: good catch! Fixed

File: modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java (hunk 1 above)
Comment (eddumelendez) [reply modules/redpanda/src/main/java/org/testcontainers/redpanda/RedpandaContainer.java:53]: waiting for https://github.com/testcontainers/testcontainers-java/pull/5741

**Review by bsideup — APPROVED** [review]
🎉

**Comment by gustavomonarin** [comment]
The current testcontainers documentation (which is awesome by the way) seems to follow a different release pipeline from library release itself, leading to an already published website that miss leads to this module to be included already in the latest_version:

```
testImplementation "org.testcontainers:redpanda:1.17.3"
```

Looking forward for the upcoming `next` release :)

**Comment by kiview** [comment]
Thanks for pointing this out @gustavomonarin. You are right, this is currently a general issue of how we publish our docs.

**Pull request merged**